assimilation_code/modules/utilities/mpi_utilities_mod.f90

```diff
@@ -576,7 +576,7 @@ subroutine send_to(dest_id, srcarray, time, label)
    if (verbose) write(*,*) "PE", myrank, ": send_to ", itemcount, " dest ", dest_id
 
    if (.not. make_copy_before_sendrecv) then
-      call MPI_Ssend(srcarray, itemcount, datasize, dest_id, tag, &
+      call MPI_Ssend(srcarray, int(itemcount,i4), datasize, dest_id, tag, &
```

Comment:
this looks ok, but there must be an mpi kind defined for the number of items.  at some point they may change it from a 4 byte int to an 8 byte int.   does:  int(itemcount, MPI_INTEGER) work? or whatever the prototype for MPI_Ssend gives for this argument?   MPI does have a lot of kinds defined for different uses.

Reply:
There is MPI_COUNT_KIND but it is MPI3.  I want to do this, but as part of updating mpi_utilities_mod to f08 and MPI3.

https://github.com/NCAR/DART/issues/261

window f08 update branch:
https://github.com/NCAR/DART/tree/mpi_f08

Also, I think we can remove the cray pointer window. 
And also, maybe (_maybe_) use fortran intrinsics for executing other programs so we don't have to use fixsystem.  Note I don't know if we can do this, especially since we do things asynchronously.